ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/processed/PackageConfigurationFactory.kt

```diff
@@ -9,34 +9,196 @@ import com.revenuecat.purchases.ui.revenuecatui.helpers.Logger
+                    availablePackages,
+                    variableDataProvider,
+                    activelySubscribedProductIdentifiers,
+                    nonSubscriptionProductIdentifiers,
+                )
+            }
+        }
+    }
+
+    @Suppress("LongParameterList", "ReturnCount")
+    private fun makeMultiTierPackageConfiguration(
+        paywallData: PaywallData,
+        packageIdsInConfig: List<String>,
+        availablePackages: List<Package>,
+        variableDataProvider: VariableDataProvider,
+        activelySubscribedProductIdentifiers: Set<String>,
+        nonSubscriptionProductIdentifiers: Set<String>,
+    ): Result<Pair<Locale, TemplateConfiguration.PackageConfiguration.MultiTier>> {
+        val tiers = paywallData.config.tiers ?: run {
+            return Result.failure(PackageConfigurationError("No tier found for $packageIdsInConfig"))
+        }
+
+        val (locale, localizedConfigurationByTier) = paywallData.tieredLocalizedConfiguration
+
+        val all = tiers.associateWith { tier ->
+            val localizationForTier = localizedConfigurationByTier[tier.id]
+                ?: return Result.failure(PackageConfigurationError("No localization found for ${tier.id}"))
+
+            val packageInfosForTier = reprocessPackagesForTiers(
+                from = availablePackages,
+                filter = tier.packageIds,
+                localization = localizationForTier,
+                variableDataProvider = variableDataProvider,
+                activelySubscribedProductIdentifiers = activelySubscribedProductIdentifiers,
+                nonSubscriptionProductIdentifiers = nonSubscriptionProductIdentifiers,
+                locale = locale,
+            )
+
+            val firstPackage = packageInfosForTier.firstOrNull()
+                ?: return Result.failure(PackageConfigurationError("No packages found for tier ${tier.id}"))
+            val defaultPackage = packageInfosForTier
+                .firstOrNull { it.rcPackage.identifier == tier.defaultPackageId } ?: firstPackage
+
+            TemplateConfiguration.PackageConfiguration.MultiPackage(
+                first = firstPackage,
+                default = defaultPackage,
+                all = packageInfosForTier,
+            )
+        }
+
+        val allTierInfos = all.entries.map { (tier, packageInfo) ->
+            val tierName = packageInfo.default.localization.tierName
+                ?: return Result.failure(
+                    PackageConfigurationError("No localized tier name found for ${tier.id}"),
+                )
+
+            TemplateConfiguration.TierInfo(
+                id = tier.id,
+                name = tierName,
+                defaultPackage = packageInfo.default,
+                packages = packageInfo.all,
+            )
+        }
+
+        return Result.success(
+            locale to TemplateConfiguration.PackageConfiguration.MultiTier(
+                firstTier = allTierInfos.first(),
```

Comment:
Just checking, but in the end we're allowing to select the default tier in a separate PR?